From 2b40a61c50b0b1762384b77ae3146976ea4cfa20 Mon Sep 17 00:00:00 2001
From: Daniel Kurtz <djkurtz@chromium.org>
Date: Wed, 27 Jan 2016 18:46:37 +0800
Subject: [PATCH 06/40] dri: Add MT21 DRI fourcc

Add MT21 DRI fourcc that corresponds to DRM_FORMAT_MT21.
---
 include/GL/internal/dri_interface.h | 1 +
 src/egl/drivers/dri2/egl_dri2.c     | 1 +
 2 files changed, 2 insertions(+)

diff --git a/include/GL/internal/dri_interface.h b/include/GL/internal/dri_interface.h
index fd3e467..4789b8c 100644
--- a/include/GL/internal/dri_interface.h
+++ b/include/GL/internal/dri_interface.h
@@ -1121,6 +1121,7 @@ struct __DRIdri2ExtensionRec {
 #define __DRI_IMAGE_FOURCC_NV12		0x3231564e
 #define __DRI_IMAGE_FOURCC_NV16		0x3631564e
 #define __DRI_IMAGE_FOURCC_YUYV		0x56595559
+#define __DRI_IMAGE_FOURCC_MT21		0x3132544D
 
 
 /**
diff --git a/src/egl/drivers/dri2/egl_dri2.c b/src/egl/drivers/dri2/egl_dri2.c
index 2bc4da0..f6c2f16 100644
--- a/src/egl/drivers/dri2/egl_dri2.c
+++ b/src/egl/drivers/dri2/egl_dri2.c
@@ -1800,6 +1800,7 @@ dri2_check_dma_buf_format(const _EGLImageAttribs *attrs)
    case DRM_FORMAT_NV21:
    case DRM_FORMAT_NV16:
    case DRM_FORMAT_NV61:
+   case DRM_FORMAT_MT21:
       plane_n = 2;
       break;
    case DRM_FORMAT_YUV410:
-- 
2.7.0.rc3.207.g0ac5344

